tests/integration/tls_tests/test_tls.py

```diff
@@ -67,18 +68,15 @@ async def test_rotate_tls_key(ops_test: OpsTest) -> None:
             ops_test, unit.name, DB_SERVICE
         )
 
-    with open(PRIVATE_KEY_PATH) as f:
-        key_contents = f.readlines()
-        key_contents = "".join(key_contents)
-
     # set external and internal key using auto-generated key for each unit
     for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
         action = await unit.run_action(action_name="set-tls-private-key")
         action = await action.wait()
         assert action.status == "completed", "setting external and internal key failed."
 
-    # sleep for long enough to process new certificate request and update certs
-    sleep(30)
+    # wait for certificate to be available and processed. Can get receive two certificate
+    # available events and restart twice so we do not wait for idle here
+    time.sleep(60)
```

Comment:
Is ok to use time.sleep() in the tests? In some other components I was using the tenacity library to avoid it.

Reply:
Great question, time.sleep() in tests is okay when it is justfied, but typically you will want to use Tenacity :)